pyfar/dsp/normalize.py

```diff
@@ -0,0 +1,151 @@
+
+    # flatten
+    
+    #input_normalized =  input_normalized.flatten() # Flattening / Reshaping to a 2d array to work with possible?
+
+    # transform data to the desired domain
+    
+    if norm_type == 'time':
+         input_normalized = input_normalized
+    
+    elif norm_type == 'abs':
+        input_normalized = np.abs(np.fft.fft(input_normalized,input_normalized.ndim - 1)) 
+    
+    elif norm_type == 'dB':
+        input_normalized = np.fft.fft(input_normalized,input_normalized.ndim - 1)
+        
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'"))
+      
+    # get bounds for normalization
+    
+    if norm_type == 'time':
+        lim = np.array([0, input_normalized.shape[1]])
+    
+    else: # if domain is abs or db
+        
+        if len(f) == 1:
+            #frequency spacing
+            df = fs/(input_normalized.shape[1])
+            
+            # get frequency limits
+            f = f/2^(1/2)
+            f = np.array([f, f*2])
+            
+            # get corresponding indicees
+            lim = np.round(f/df) + 1
+            
+        else: 
+            # frequency spacing
+            df = fs/(input_normalized.shape[1])
+            
+            # get corresponding indices
+            lim = np.round(f/df) + 1
+              
+    # get values for normalization
+    
+    if operation == 'max':
+        if norm_type == 'time':
+            values = np.max(np.abs(input_normalized[:,lim[0]:lim[1]]),input_normalized.ndim - 1)
+        else: 
+            values = np.max(input_normalized[:,lim[0]:lim[1]],input_normalized.ndim - 1)
+    
+    elif operation == 'mean':
+        values = np.mean(input_normalized[:,lim[0]:lim[1]],input_normalized.ndim - 1)
+    
+    elif operation == 'rms':
+        values = np.sqrt(np.mean(((input_normalized[:,lim[0]:lim[1]])**2),1),input_normalized.ndim - 1)
+    
+    else: 
+        raise ValueError(("'operation' must be 'abs', 'mean' or 'rms'"))
+        
+    
+    if norm_type == 'db':
+        valuesLin = 10^(values/20)
+        valueLin = 10^(value/20)
+    
+    else:
+        valuesLin = values
+        valueLin = value
+        
+    # apply normalization
+     
+    if channelwise == 'each':
+        input_normalized = np.divide(input_normalized.T,valuesLin).T
+
+    elif channelwise == 'max':
+        input_normalized = input_normalized / np.max(valuesLin)
+        
+    elif channelwise == 'min':
+        input_normalized = input_normalized / np.min(valuesLin)
+    
+    elif channelwise == 'mean':
+        input_normalized = input_normalized / np.mean(valuesLin)
+    else: 
+        raise ValueError(("channelwise must be 'each', 'max', 'min' or  'mean'"))
+    
+    # normalize to value
+                                                       
+    input_normalized = input_normalized * valueLin
+    
+    # reshape
+    
+    #input_normalized = input_normalized.reshape(input_signal.cshape)
+    
+    return input_normalized
```

Comment:
This returns a numpy array, but should return a signal. One possibility would be
```
signal_normalized = signal.copy()
signal_normalized.time = input_normalized
```